src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -1091,6 +1102,40 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
                 @. ∂ᶜρq_tot_err_∂ᶜq_totʲ =
                     -(dtγ * ᶜadvdivᵥ_matrix() ⋅ ∂ᶜupdraft_mass_flux_∂ᶜqtot)
 
+                # ∂/∂u₃ʲ contributions to grid-mean ρe_tot and ρq_tot from updraft mass flux
+                ∂ᶜρe_tot_err_∂ᶠu₃ʲ =
+                    matrix[@name(c.ρe_tot), @name(f.sgsʲs.:(1).u₃)]
+                @. ∂ᶜρe_tot_err_∂ᶠu₃ʲ =
+                    dtγ * ᶜadvdivᵥ_matrix() ⋅
+                    DiagonalMatrixRow((ᶠinterp(ᶜρʲs.:(1) * ᶜJ) / ᶠJ),) ⋅
+                    DiagonalMatrixRow(ᶠinterp(Y.c.sgsʲs.:(1).ρa / ᶜρʲs.:(1))) ⋅
+                    DiagonalMatrixRow(g³³(ᶠgⁱʲ)) ⋅ ᶠinterp_matrix() ⋅
```

Comment:
```suggestion
                    DiagonalMatrixRow(ᶠinterp(ᶜρʲs.:(1) * ᶜJ) / ᶠJ * ᶠinterp(Y.c.sgsʲs.:(1).ρa / ᶜρʲs.:(1)) * g³³(ᶠgⁱʲ)) ⋅ ᶠinterp_matrix() ⋅
```
Try to minimize the number of times you use `⋅`, since it is a much more expensive operation than `*` (both in terms of runtime and compilation). Whenever you have multiple diagonal matrices in sequence, it is better to combine them into one diagonal matrix.